x-pack/plugins/inference/common/chat_complete/tool_schema.ts

```diff
@@ -35,28 +35,18 @@ interface ToolSchemaTypeNumber extends ToolSchemaFragmentBase {
   enum?: string[] | readonly string[];
 }
 
-interface ToolSchemaAnyOf extends ToolSchemaFragmentBase {
-  anyOf: ToolSchemaType[];
-}
-
-interface ToolSchemaAllOf extends ToolSchemaFragmentBase {
-  allOf: ToolSchemaType[];
-}
```

Comment:
As discussed with @dgieselaar: Gemini doesn't support type composition for tool definition, so we can't easily have it on the inference API. But given that the feature doesn't seem that useful, it feels like a very acceptable removal.